File: Jawa/Code/Doctor.java
import java.util.ArrayList;

public class Doctor extends Person {
    private String specialization;
    private ArrayList<Schedule> schedules;

    public Doctor(String name, String specialization) {
        super(name);
        this.specialization = specialization;
        this.schedules = new ArrayList<>();
    }

    //get set
    public void setSpecialization(String specialization) {
        this.specialization = specialization;
    }

    public String getSpecialization() {
        return specialization;
    }

    //fungsi add shedule
    public void addSchedule(Schedule s) {
        schedules.add(s);
    }

    //fungsi print doctor
    public void printDoctor() {
        System.out.println("Doctor: " + getName() + ", Specialization: " + specialization);
        System.out.println("Schedules:");
        if (schedules.isEmpty()) {
            System.out.println("- No schedule available.");
        } else {
            for (Schedule s : schedules) {
                s.printSchedule();
            }
        }
        System.out.println("================================================");
    }
}

File: Uler/Code/Doctor.py
from Person import Person
from Schedule import Schedule

class Doctor(Person):
    def __init__(self, name, specialization):
        super().__init__(name)
        self.specialization = specialization
        self.schedules = []

    #get set
    def set_specialization(self, specialization):
        self.specialization = specialization

    def get_specialization(self):
        return self.specialization

    #fungsi add schedule
    def add_schedule(self, schedule: Schedule):
        self.schedules.append(schedule)

    #fungsi print dokter
    def print_doctor(self):
        print(f"Doctor: {self.get_name()}, Specialization: {self.specialization}")
        print("Schedules:")
        if not self.schedules:
            print("- No schedule available.")
        else:
            for s in self.schedules:
                s.print_schedule()
        print("================================================")